sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/logging_handler.py

```diff
@@ -51,308 +44,99 @@
-#         status = _http_send(self._logger, data_to_send, self.service_endpoint_uri, self.send_timeout)
-#         if status is SUCCESS:
-#             self.consecutive_failures = 0
-#             return
-#         self.consecutive_failures = self.consecutive_failures + 1
-
-#         if self.consecutive_failures <= self.retry:
-#             for data in data_to_send:
-#                 self._queue.put(data)
-
-
-# class _RetryAsynchronousSender(AsynchronousSender):
-#     """Asynchronous sender with limited retry.
-
-#     SenderBase does infinite retry; this class avoids that.
-#     """
-
-#     def __init__(self, logger, timeout=10, retry=3):
-#         super(_RetryAsynchronousSender, self).__init__()
-#         self._logger = logger
-#         self.send_timeout = timeout
-#         self.retry = retry
-#         self.consecutive_failures = 0
-
-#     def send(self, data_to_send):
-#         """Override the default resend mechanism in SenderBase.
-
-#         Stop resend based on retry during failure.
-#         """
-#         status = _http_send(self._logger, data_to_send, self.service_endpoint_uri, self.send_timeout)
-#         if status is SUCCESS:
-#             self.consecutive_failures = 0
-#             return
-#         self.consecutive_failures = self.consecutive_failures + 1
-
-#         if self.consecutive_failures <= self.retry:
-#             for data in data_to_send:
-#                 self._queue.put(data)
-
-
-def _json_serialize_unknown(obj):
-    """JSON serializer for objects not serializable by default json code.
-
-    :param obj: the object to be serialized
+def in_jupyter_notebook() -> bool:
     """
-    if isinstance(obj, datetime.datetime):
-        return obj.isoformat()
-    raise TypeError("Type %s not serializable" % type(obj))
-
+    Checks if user is using a Jupyter Notebook. This is necessary because logging is not allowed in
+    non-Jupyter contexts.
 
-def _http_send(logger, data_to_send, service_endpoint_uri, send_timeout=10):
-    """Replicate Application Insights SenderBase send method.
-
-    :param logger: logger
-    :param data_to_send: items to send
-    :param service_endpoint_uri: endpoint
-    :param send_timeout: timeout
-    :return: SUCCESS/FAILURE
+    Adapted from https://stackoverflow.com/a/22424821
+    """
+    try:  # cspell:ignore ipython
+        from IPython import get_ipython
+        if 'IPKernelApp' not in get_ipython().config:
+            return False
+    except ImportError:
+        return False
+    except AttributeError:
+        return False
+    return True
+
+
+def get_appinsights_log_handler(
+     user_agent,
+     *args, # pylint: disable=unused-argument
+     instrumentation_key=None,
+     component_name=None,
+     **kwargs
+ ):
+    """Enable the OpenCensus logging handler for specified logger and instrumentation key to send info to AppInsights.
+
+    Enable diagnostics collection with the :func:`azureml.telemetry.set_diagnostics_collection` function.
```

Comment:
Is this legacy comment from v1 SDK?

Reply:
Yes, good catch. I removed it and updated the docstring.